aspnetcore/tutorials/signalr/sample/wwwroot/js/chat.js

```diff
@@ -10,15 +10,16 @@ connection.on("ReceiveMessage", function (user, message) {
     document.getElementById("messagesList").appendChild(li);
 });
 
-connection.start().catch(function (err) {
+connection.start().then(function(){
+    document.getElementById("sendButton").addEventListener("click", function (event) {
```

Comment:
It would probably be better to just set the button as unclickable until `.then` is called

Reply:
super nit: It probably looks cleaner if this was moved out of the `.then`, but I don't care enough to fight it.